rep-2000.rst

```diff
@@ -823,6 +823,120 @@ Package manager use for dependencies:
++==============+==================+===============+==================+============+=================+================+
+|    amd64     | Tier 1 [d][a][s] | Tier 1 [a][s] | Tier 2 [d][a][s] | Tier 3 [s] |   Tier 3 [s]    |                |
++--------------+------------------+---------------+------------------+------------+-----------------+----------------+
+|    arm64     | Tier 1 [d][a][s] |               |                  |            |   Tier 3 [s]    |   Tier 3 [s]   |
++--------------+------------------+---------------+------------------+------------+-----------------+----------------+
+|    arm32     |    Tier 3 [s]    |               |                  |            |   Tier 3 [s]    |   Tier 3 [s]   |
++--------------+------------------+---------------+------------------+------------+-----------------+----------------+
+
+The following indicators show what delivery mechanisms are available for each platform.
+
+" [d] " Distribution-specific (Debian, RPM, etc.) packages will be provided for this platform for packages submitted to the rosdistro.
+
+" [a] " Binary releases are provided as a single archive per platform containing all packages in the Galactic ROS 2 repos file [10]_.
+
+" [s] " Compilation from source.
+
+Middleware Implementation Support:
+
++--------------------------+---------------------+---------------+----------------------------+--------------------------------+
+| Middleware Library       | Middleware Provider | Support Level | Platforms                  | Architectures                  |
++==========================+=====================+===============+============================+================================+
+| rmw_cyclonedds_cpp       | Eclipse Cyclone DDS | Tier 1        | All Platforms              | All Architectures              |
++--------------------------+---------------------+---------------+----------------------------+--------------------------------+
+| rmw_fastrtps_cpp         | eProsima Fast-DDS   | Tier 1        | All Platforms              | All Architectures              |
++--------------------------+---------------------+---------------+----------------------------+--------------------------------+
+| rmw_connextdds           | RTI Connext         | Tier 1        | Ubuntu, Windows, and macOS | All Architectures except arm64 |
++--------------------------+---------------------+---------------+----------------------------+--------------------------------+
+| rmw_fastrtps_dynamic_cpp | eProsima Fast-DDS   | Tier 2        | All Platforms              | All Architectures              |
++--------------------------+---------------------+---------------+----------------------------+--------------------------------+
+
+Middleware implementation support is dependent upon the platform support tier.
+For example a Tier 1 middleware implementation on a Tier 2 platform can only receive Tier 2 support.
+
+Minimum language requirements:
+
+- C++17
```

Comment:
Is there any reason not to target c++20?

Reply:
> Is there any reason not to target c++20?

There are a few reasons we don't think we are going to do this for Humble:

1. Every time we've switched C++ versions, it has taken some non-trivial amount of work to keep things working without warnings.  Someone would have to step up to do that work.
1. All of our Tier 1 platforms probably support C++20 well, but not all of the Tier 2 and Tier 3 ones do.
1. Part of the ROS 2 community deals with safety certification, things like MISRA.  Since those standards tend to lag behind the relevant standards, switching to a new C++ version too early can leave those users behind.